theano/sandbox/rng_mrg.py

```diff
@@ -1292,8 +1292,17 @@ def uniform(self, size, low=0.0, high=1.0, ndim=None, dtype=None,
             The output data type. If dtype is not specified, it will be
             inferred from the dtype of low and high, but will be at
             least as precise as floatX.
-
+        nstreams:
+            The number of streams to use for sampling the random numbers
+            in parallel.
+        broadcastable:
```

Comment:
In the issue, I told I wasn't sure of the interface. Was there other discussion about that that I missed? I'm still not sure this is the right way to do it. We don't add extra op in the graph at other place when the user build the graph himself to make the broadcast pattern the one he wants.

What about changing mrg_uniform_base.make_node try to infer the broadcastable flags? If we can know that some of the shapes will be 1 (if they are constant for example). This would be more user friendly, they won't need to specify it manually themself. In the make_node, we could do:
```
for i in ndim:
   if extract_constant(size[[i]) == 1:
      br = True
   else: 
      br = False
```
There is still the case when the input size is a vector that the user provide. In that case, we could just let him add the petternbroadcast himself or keep what this PR does. @dwf it is you that talked about this. If we change the make_node as discussed, would you still benefit from the broadcastable parameter?

@lamblin @abergeron